vignettes/demographic_data.Rmd

```diff
@@ -0,0 +1,257 @@
+  x = reorder(departamento, -proportion_si),
+  y = proportion_si
+)) +
+  geom_bar(stat = "identity", fill = "#10bed2", color = "black", width = 0.6) +
+  labs(
+    title = "Proportion of dwellings with access to WSS by department",
+    x = "Department",
+    y = "Proportion"
+  ) +
+  theme_minimal() +
+  theme(
+    axis.text.x = element_text(angle = 45, hjust = 1),
+    plot.title = element_text(hjust = 0.5)
+  )
+```
+
+## Data Integration: Combining Datasets for Enhanced Analysis
+
+::: {style="text-align: justify;"}
+Also, with the CNPV data, we could merge data from the different categories and compare information. For example, we would download a dataset from the persons demographic category and another one from persons social. The first one presents information regarding population's literacy and the second one regarding its marital status. The following datasets might take a while to download.
+:::
+
+### Data load
+
+```{r more data load, include=TRUE, echo=TRUE}
+literacy <- download_demographic("DANE_CNPVPD_2018_16PM")
+marital_status <- download_demographic("DANE_CNPVPS_2018_7PM")
+```
+::: {style="text-align: justify;"}
+For this example, we will focus on a specific municipality, examining the total area and specifically concentrating on the age group between 15 and 24 years old. Since we want to compare both datasets simultaneously, it's important to first ensure they have the same population count under identical conditions. Most columns in the demographic datasets contain a [total]{.underline} value, which aggregates data across all conditions. For instance, the "sex" column includes "male", "female", and ["total"]{.underline}, representing both genders combined. Similarly, the "marital status" column's ['total']{.underline} value encompasses the entire population, irrespective of their marital status.
+:::
+
+### Data check
+
+::: {style="text-align: justify;"}
+Since the age group in literacy is complete, we can just filter using the desired age range and the total counts for other variables, including area, sex and ability to read.
+:::
+
+```{r literacy}
+lit <- literacy %>%
+  filter(
+    municipio == "Villapinzón",
+    grupo_de_edad == "entre_15_y_24_anos",
+    area == "total",
+    sabe_leer_y_escribir == "total",
+    sexo == "total"
+  ) %>%
+  select(sexo, total)
+
+head(lit)
+```
+
+::: {style="text-align: justify;"}
+We will now filter marital status for people between 15 and 24 years. This dataset age groups are divided from 15 to 19 and from 20 to 24 years old, so we will need to merge the two groups. To do this, we can group by a common variable, like sex and add as follows:
+:::
+
+```{r}
+ms_15_24 <- marital_status %>%
+  filter(
+    municipio == "Villapinzón",
+    area == "total",
+    grupo_de_edad %in% c("15_a_19", "20_a_24"),
+    estado_civil == "total",
+    sexo == "total"
+  ) %>%
+  group_by(sexo) %>%
+  summarise(total = sum(total)) %>%
+  select(sexo, total)
+
+head(ms_15_24)
+```
```

Comment:
We had looked into it at various times, and not for all the cases it's possible. Many datasets present a similar format regarding the age group but there are also many that are really different, and there isn't enough data to achieve the harmonization. We would have to assume a lot of things and also since the data comes like this, we think it is better not to modify that much.